web-ui/src/components/member_selector/member_selector_dialog/MemberSelectorDialog.jsx

```diff
@@ -230,6 +254,22 @@ const MemberSelectorDialog = ({
         member => !selectedMembers.includes(member)
       );
 
+      // Exclude members that don't have the selected tenure.
+      if (tenure === Tenures.Custom) {
```

Comment:
If a custom tenure date was selected, keep only the members whose start date was on or before that date.